### ISSUE TITLE
Supported way of setting hosts on external-request segments?

### ISSUE THREAD

**Issue by jdelStrother** [newrelic/newrelic-ruby-agent#2655]
### Is your feature request related to a problem? Please describe.

Our app fetches a lot of RSS feeds from a lot of different third-party hosts, which generates a lot of metric names like `External/foo.com/Net::HTTP/GET`, which can lead to [MGI](https://docs.newrelic.com/docs/new-relic-solutions/solve-common-issues/troubleshooting/metric-grouping-issues/)

### Feature Description

I'd like a way of grouping those requests, while still allowing all our other HTTP requests (eg API calls to s3.amazonaws.com) to keep their own default grouping.  Perhaps something like:

```ruby
NewRelic::Agent.with_external_naming_override("rss-host") do
   URI.open("https://foo.com/a.rss")
   URI.open("https://bar.com/b.rss")
end
```
which would generate two segments, both with the grouped name `External/rss-host/Net::HTTP/GET`.

(I can definitely see why that exact API has issues, open to alternative suggestions)

### Describe Alternatives

You could maybe do this with NewRelic's metric-normalization rules, but wanting to keep existing names for things like our API services makes it harder - we'd need to know in advance the hosts that we want to whitelist, and then use a REPLACE rule on all the remaining ones.

I'm considering monkey-patching `ExternalRequestSegment#name` so that it can be overridden by a thread-local variable.

### Priority
Really Want

**Comment by jdelStrother** [comment]
> I would also like to point out that there is [another metric](https://github.com/newrelic/newrelic-ruby-agent/blob/dev/lib/new_relic/agent/transaction/external_request_segment.rb#L224) other than the segment name that gets created using the host

Good catch, thanks. I'll try overriding `ExternalRequestSegment#host` instead.

**Comment by tannalynn** [comment]
Hello, sorry for the confusion!

Our github issues are linked to Jira, and due to some changes made to feature requests on Jira recently, we had several of our github issues autoclose, including this one. We have a list of issues that were closed that we plan to review, but we haven't done that yet. I'll reopen this now though.
